packages/iov-lisk/src/liskclient.ts

```diff
@@ -0,0 +1,200 @@
+import axios from "axios";
+import { ReadonlyDate } from "readonly-date";
+import { Stream } from "xstream";
+
+import {
+  Address,
+  BcpAccount,
+  BcpAccountQuery,
+  BcpAddressQuery,
+  BcpNonce,
+  BcpQueryEnvelope,
+  BcpTicker,
+  BcpTransactionResponse,
+  ConfirmedTransaction,
+  IovReader,
+  Nonce,
+  TokenTicker,
+} from "@iov/bcp-types";
+import { Encoding } from "@iov/encoding";
+import { Algorithm, ChainId, PostableBytes, PublicKeyBytes, Tag, TxId, TxQuery } from "@iov/tendermint-types";
+
+import { Parse } from "./parse";
+
+function isAddressQuery(query: BcpAccountQuery): query is BcpAddressQuery {
+  return (query as BcpAddressQuery).address !== undefined;
+}
+
+/**
+ * Encodes the current date and time as a nonce
+ */
+export function generateNonce(): Nonce {
+  const now = new ReadonlyDate(ReadonlyDate.now());
+  return Parse.timeToNonce(now);
+}
+
+export class LiskClient implements IovReader {
+  private readonly baseUrl: string;
+
+  constructor(baseUrl: string) {
+    this.baseUrl = baseUrl;
+  }
+
+  public disconnect(): void {
+    // no-op
+  }
+
+  public async chainId(): Promise<ChainId> {
+    const url = this.baseUrl + "/api/node/constants";
+    const result = await axios.get(url);
+    const responseBody = result.data;
+    return responseBody.data.nethash;
+  }
+
+  public async height(): Promise<number> {
+    const url = this.baseUrl + "/api/node/status";
+    const result = await axios.get(url);
+    const responseBody = result.data;
+    return responseBody.data.height;
+  }
+
+  public async postTx(bytes: PostableBytes): Promise<BcpTransactionResponse> {
+    const transactionId = JSON.parse(Encoding.fromUtf8(bytes)).id as string;
```

Comment:
nice check, that this is really a lisk transaction. I see JSON.parse throwing an error on lots of bad data